optuna/integration/allennlp.py

```diff
@@ -163,16 +360,44 @@ class AllenNLPPruningCallback(EpochCallback):
         monitor:
             An evaluation metric for pruning, e.g. ``validation_loss`` or
             ``validation_accuracy``.
+
     """
 
-    def __init__(self, trial: optuna.trial.Trial, monitor: str):
+    def __init__(
+        self,
+        trial: Optional[optuna.trial.Trial] = None,
+        monitor: Optional[str] = None,
+    ):
         _imports.check()
 
         if allennlp.__version__ < "1.0.0":
             raise Exception("AllenNLPPruningCallback requires `allennlp`>=1.0.0.")
 
-        self._trial = trial
-        self._monitor = monitor
+        if trial is not None and monitor is not None:
+            self._trial = trial
+            self._monitor = monitor
+        else:
+            _environment_variables = _get_environment_variables_for_trial()
+            study_name = _environment_variables["study_name"]
+            trial_id = _environment_variables["trial_id"]
+            monitor = _environment_variables["monitor"]
+            storage = _environment_variables["storage"] or None
+
+            if (
+                study_name is not None
+                and trial_id is not None
+                and monitor is not None
+                and storage is not None
+            ):
+                study = load_study(study_name, storage, pruner=_create_pruner())
+                self._trial = Trial(study, int(trial_id))
+                self._monitor = monitor
+            else:
+                message = (
+                    "Fail to load study.\n"
+                    "AllenNLPPruningCallback works only with Optuna and RDB or Redis storages."
+                )
+                raise Exception(message)
```

Comment:
Can we provide more information and finer grained error handling by using a more concrete type of error?
```suggestion
                raise RuntimeException(message)
```